src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -785,17 +747,6 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
 
     if (fConstructor)
     {
-        // If we are invoking a constructor on an array then we must
-        // handle this specially.  String objects allocate themselves
-        // so they are a special case.
-        if (ownerType.IsArray()) {
-            gc.retVal = InvokeArrayConstructor(ownerType,
-                                               pMeth,
-                                               objs,
-                                               gc.pSig->NumFixedArgs());
-            goto Done;
-        }
-
         MethodTable * pMT = ownerType.AsMethodTable();
 
         {
```

Comment:
We still have this special check for string constructors here. It makes me wonder whether it is an actual improvement to move the special check for array constructors to a different spot. I think it made more logical sense to have both these special checks together.

(I do not have a strong opinion on this.)

Reply:
I'll move it back. The more I look into this less valuable this seems. I am going to consider this PR for fixing the widening logic and adding some much needed test coverage for Reflection scenarios.